src/common/handler.ts

```diff
@@ -0,0 +1,84 @@
+import { Connection } from '../connection/connection'
+import { Wallet } from 'ethers'
+import { Reference, RequestOptions } from '@ethersphere/bee-js'
+import { prepareEthAddress } from '../utils/address'
+import { getUnixTimestamp } from '../utils/time'
+import { getFeedData, writeFeedData } from '../feed/api'
+import { getRawDirectoryMetadataBytes } from '../directory/adapter'
+import { DIRECTORY_TOKEN, FILE_TOKEN } from '../file/handler'
+import { LookupAnswer } from '../feed/types'
+import { assertRawDirectoryMetadata, combine } from '../directory/utils'
+import { RawDirectoryMetadata } from '../pod/types'
+
+/**
+ * Add child file or directory to defined parent directory
+ *
+ * @param connection current connection
+ * @param wallet wallet of the pod
+ * @param parentPath parent path
+ * @param entryPath entry path
+ * @param isFile define if entry is file or directory
```

Comment:
Dynamically obtaining the file type creates an unnecessary delay when receiving data over the network.
Since this method is used only inside the project, it is most likely that it can be left it in the same form.